api/src/main/java/org/pmiops/workbench/interceptors/AuthInterceptor.java

```diff
@@ -112,17 +121,47 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
       return false;
     }
 
-    // TODO: get token info and check that as well
-
     // TODO: check Google group membership to ensure user is in registered user group
 
-    SecurityContextHolder.getContext().setAuthentication(new UserAuthentication(userInfo, token));
+    String userEmail;
+    String gsuiteDomainSuffix =
+        "@" + workbenchConfigProvider.get().googleDirectoryService.gSuiteDomain;
+    if (!userInfo.getEmail().endsWith(gsuiteDomainSuffix)) {
+      try {
+        // If the email isn't in our GSuite domain, try FireCloud; we could be dealing with a
+        // pet service account. In both AofU and FireCloud, the pet SA is treated as if it were
+        // the user it was created for.
+        userEmail = fireCloudService.getMe().getUserInfo().getUserEmail();
+      } catch (ApiException e) {
+        log.log(Level.INFO, "FireCloud lookup for {0} failed, can't access the workbench: {2}",
+            new Object[]{userInfo.getEmail(), e.getMessage()});
+        response.sendError(HttpServletResponse.SC_FORBIDDEN);
+        return false;
+      }
+    } else {
+      userEmail = userInfo.getEmail();
```

Comment:
nit: Just initialize String userEmail with this above, instead of using the else

Reply:
Done.